src/domain/accounts/domain/src/services/account_service.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashMap;
+
+use database_common::{EntityPageListing, PaginationOpts};
+use internal_error::InternalError;
+use opendatafabric as odf;
+use thiserror::Error;
+
+use crate::{Account, AccountPageStream};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// TODO: Private Datasets: merge with AuthenticationService?
+// TODO: Private Datasets: tests
+#[async_trait::async_trait]
+pub trait AccountService: Sync + Send {
+    // TODO: Private Datasets: extract to AccountRegistry?
+    fn all_accounts(&self) -> AccountPageStream;
+
+    async fn list_all_accounts(
+        &self,
+        pagination: PaginationOpts,
+    ) -> Result<EntityPageListing<Account>, ListAccountError>;
```

Comment:
Similarly, very dangerous operations

Reply:
Since only rebac-indexer uses it, maybe it should inject a repository directlly.

Reply:
The `list_all_accounts()` method is absorbed by `all_accounts()` and moved to `ExpensiveAccountRepositoryExt`

Done